internal/ir/xds.go

```diff
@@ -393,14 +394,24 @@ func (r RouteDestination) Validate() error {
 }
 
 // NewRouteDest creates a new RouteDestination.
-func NewRouteDest(host string, port uint32, weight uint32) *RouteDestination {
+func NewRouteDest(host string, port uint32) *RouteDestination {
 	return &RouteDestination{
-		Host:   host,
-		Port:   port,
-		Weight: weight,
+		Host: host,
+		Port: port,
 	}
 }
 
+func NewRouteDestWithWeight(host string, port uint32, weight uint32) *RouteDestination {
+	rd := &RouteDestination{
+		Host: host,
+		Port: port,
+	}
+	if weight != 0 {
```

Comment:
imo this check isn't needed, what if the caller wants to explicitly set the weight to zero ?

Reply:
Addressed in the new commit.